Integrations/SplunkPy/SplunkPy.py

```diff
@@ -489,6 +489,60 @@ def splunk_submit_event_command():
         demisto.results('Event was created in Splunk index: ' + r.name)
 
 
+def splunk_submit_event_hec(hec_token, baseurl, event, fields, host, index, source_type, source, time_):
+
+    if hec_token is None:
+        raise Exception('The HEC Token was not provided')
+
+    args = {}
+    args['event'] = event
+    if fields:
+        args['fields'] = {
+            'fields': fields
+        }
+    if host:
+        args['host'] = host
+    if index:
+        args['index'] = index
+    if source_type:
+        args['sourcetype'] = source_type
+    if source:
+        args['source'] = source
+    if time_:
+        args['time'] = time_
+
+    headers = {
+        'Authorization': 'Splunk %s' % hec_token,
+        'Content-Type': 'application/json'
+    }
+
+    response = requests.post(baseurl + '/services/collector/event', data=json.dumps(args), headers=headers,
+                             verify=VERIFY_CERTIFICATE)
+    return response
+
+
+def splunk_submit_event_hec_command():
+
+    hec_token = demisto.params().get('hec_token')
+    hec_port = demisto.params().get('hec_port')
```

Comment:
In order to prevent connection exceptions let's also make sure the port is also defined```
```
if not hec_port:
    raise Exception(...)
```